htlcswitch/switch.go

```diff
@@ -1032,11 +1032,19 @@ func (s *Switch) extractResult(deobfuscator ErrorDecrypter, n *networkResult,
 	case *lnwire.UpdateFailHTLC:
 		// TODO(yy): construct deobfuscator here to avoid creating it
 		// in paymentLifecycle even for settled HTLCs.
-		paymentErr := s.parseFailedPayment(
+		encryptedErr, paymentErr := s.parseFailedPayment(
 			deobfuscator, attemptID, paymentHash, n.unencrypted,
 			n.isResolution, htlc,
 		)
 
+		// The failure is opaque and needs to be deferred.
+		if encryptedErr != nil {
```

Comment:
length check instead maybe?

Reply:
it's a bit of an anti pattern to do anything with a return value before first checking the error returned by a function. 

so what do you think about swapping these instead?

```
        if paymentErr != nil {
			return &PaymentResult{
				Error: paymentErr,
			}, nil
		}

		return &PaymentResult{
			EncryptedError: encryptedErr,
		}, nil
```

Reply:
other option, is to actually switch around the return values of `parseFailedPayment` so that the error is actually returned first. Cause usually, in a general  function, an `error` return value indicates that something went wrong in the method call itself. But here that isnt really the case, it is more a conversion from a set of info into an `error` type. 

So id say it makes sense change up the return values to indicate this. So either:

```
error, []byte 
or
(fn.Option[error], []byte)
```

what  is also nice about this is that if one day this method is updated in a way that it can itself fail, then it can still add an `error` return value at the end of the return list & it will be more clear which error is for what